src/panels/lovelace/cards/hui-thermostat-card.ts

```diff
@@ -114,13 +104,19 @@ export class HuiThermostatCard extends LitElement implements LovelaceCard {
     }
 
     const mode = stateObj.state in modeIcons ? stateObj.state : "unknown-mode";
+
+    if (!this._radius || this._radius === 0) {
+      this._radius = 100;
+    }
+
     return html`
-      ${this.renderStyle()}
       <ha-card
         class="${classMap({
           [mode]: true,
-          large: this._broadCard!,
-          small: !this._broadCard,
+          large: this._large!,
+          medium: this._medium!,
+          small: this._small!,
```

Comment:
Am I correct that those values are only calculated once, and only used here?
If so, can't the calculation be made here too?

Reply:
Ok. I found the other place _initialLoad is called from.
But is *that* still necessary? Does it need to be called from both firstUpdated and connectedCallback?

Reply:
We can only get the size when it is connected, and we don't know which will fire first, connected or updated.

Reply:
Ah yes.